hypothesis-python/src/hypothesis/provisional.py

```diff
@@ -142,6 +142,24 @@ def domains(
     )
 
 
+# Strategy used to generate URL fragments (e.g. "#foo"); extracted to top-level
+# so that it can be reliably tested.
```

Comment:
I think it's worth being explict that this is "so that the lambda is only defined once, which ensures that test coverage is not flaky.  See `#2995`."

Reply:
Does the “only defined once” part affect coverage? I'm under the impression that it doesn't.

(I pulled it out so that I could write tests that bypass `urls()`; the fact that the lambda only gets defined once is just a side-effect.)

Reply:
I don't know, actually!  Thinking it through I'd expect not.

How about "so that it can be tested independently of the `urls()` strategy." then?